frontend/packages/dev-console/src/components/topology/__tests__/TopologyPage.spec.tsx

```diff
@@ -0,0 +1,114 @@
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    expect(wrapper.find(AsyncComponent).exists()).toBe(true);
+  });
+
+  it('should render topology graph page', () => {
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    expect(wrapper.find(ConnectedTopologyDataController).exists()).toBe(true);
+  });
+
+  it('should render projects list page', () => {
+    topologyProps.match.params.name = '';
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    expect(wrapper.find(ProjectListPage).exists()).toBe(true);
+  });
+
+  it('should render view shortcuts button on topology page toolbar', () => {
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    const namespacesPageWrapper = wrapper.find(NamespacedPage).shallow();
+    expect(namespacesPageWrapper.find('[data-test-id="topology-view-shortcuts"]').exists()).toBe(
+      true,
+    );
+  });
+
+  it('should not render view shortcuts button on topology list page toolbar', () => {
+    topologyProps.match.path = '/topology/ns/topology-test/list';
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    const namespacesPageWrapper = wrapper.find(NamespacedPage).shallow();
+    expect(namespacesPageWrapper.find('[data-test-id="topology-view-shortcuts"]').exists()).toBe(
+      false,
+    );
+  });
+
+  it('should show the topology icon when on topology list page', () => {
+    topologyProps.match.path = '/topology/ns/topology-test/list';
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    const namespacesPageWrapper = wrapper.find(NamespacedPage).shallow();
+    expect(namespacesPageWrapper.find(Tooltip).props().content).toBe('Topology View');
+    expect(namespacesPageWrapper.find(Link).props().to).toBe('/topology/ns/topology-test/graph');
+  });
+
+  it('should show the topology list icon when on topology page', () => {
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    const namespacesPageWrapper = wrapper.find(NamespacedPage).shallow();
+    expect(namespacesPageWrapper.find(Tooltip).props().content).toBe('List View');
+    expect(namespacesPageWrapper.find(Link).props().to).toBe('/topology/ns/topology-test/list');
+  });
+
+  it('should render topology when workload is loaded', () => {
+    const Component = () => renderTopology(renderTopologyProps);
+    const wrapper = shallow(<Component />);
+    expect(wrapper.find(StatusBox).exists()).toBe(true);
+    expect(wrapper.find(Topology).exists()).toBe(true);
+  });
+
+  it('should render all projects list page for graph view', () => {
+    topologyProps.match.params.name = '';
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    const namespacesPageWrapper = wrapper.find(NamespacedPage).shallow();
+    expect(namespacesPageWrapper.find(Link).props().to).toBe('/topology/all-namespaces/list');
+  });
+
+  it('should render all projects list page for graph view', () => {
```

Comment:
this test case title is same as the previous one

Reply:
Good catch, I missed that lol. These tests have gone through quite a few changes, I was focusing on the content :D Glad there are two reviewers here.